services/entitlement/service/src/messaging/handlers/entitlement-authenticated-message-handler.ts

```diff
@@ -1,48 +1,65 @@
-import { buildAuthPgSettings } from '@axinom/mosaic-db-common';
 import {
-  AuthenticatedManagementSubject,
-  authenticationMiddleware,
-  getMessageInfoManagementSubject,
+  buildAuthPgSettings,
+  setPgSettingsConfig,
+} from '@axinom/mosaic-db-common';
+import {
+  getAuthenticatedManagementSubject,
+  GuardedContext,
+  IdGuardErrors,
 } from '@axinom/mosaic-id-guard';
+import { MessagingSettings } from '@axinom/mosaic-message-bus-abstractions';
+import { Logger, MosaicError } from '@axinom/mosaic-service-common';
 import {
-  MessageHandler,
-  MessageInfo,
-  OnMessageMiddleware,
-} from '@axinom/mosaic-message-bus';
-import { Dict } from '@axinom/mosaic-service-common';
-import { SubscriptionConfig } from 'rascal';
+  TransactionalInboxMessageHandler,
+  TypedTransactionalMessage,
+} from '@axinom/mosaic-transactional-inbox-outbox';
+import { DatabaseClient } from 'pg-transactional-outbox';
 import { Config } from '../../common';
 
 /**
- * Guard a message handler by getting and verifying the JWT token. It changes
- * the `MessageInfo` parameter to `AuthenticatedMessageInfo` which contains the
- * JWT token subject.
- * In addition it checks that the subjects permission match the required ones.
+ * Custom channel message handler to verify channel permissions
  */
-export abstract class EntitlementAuthenticatedMessageHandler<
-  TContent,
-> extends MessageHandler<TContent> {
+export abstract class EntitlementAuthenticatedTransactionalMessageHandler<
+  T,
+> extends TransactionalInboxMessageHandler<T, Config> {
   constructor(
-    messagingKey: string,
-    protected readonly config: Config,
-    overrides?: SubscriptionConfig,
-    middleware: OnMessageMiddleware[] = [],
+    messagingSettings: MessagingSettings,
+    logger: Logger,
+    messageProducerService:
+      | 'ax-monetization-grants-service'
```

Comment:
Just to be on the safe side, I would use something like
`MonetizationGrantsServiceMultiTenantMessagingSettings.ClaimSetPublished.serviceId`
here instead of just string.